### PR TITLE
[18.0][MIG] bi_sql_editor: Migration to 18.0

### PR DESCRIPTION
**Pull request by thienvh332**
### Ported PRs
- #620
- #805
### Depend on:
- https://github.com/oca/reporting-engine/pull/937

### CODE REVIEW

File: bi_sql_editor/models/bi_sql_view_field.py
Comment (henrybackman) [inline bi_sql_editor/models/bi_sql_view_field.py]: According to the migration guide the access to "ir.model.*" methods is removed in 15:
- https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-15.0
but is this still a valid use? Not familiar with this so just wanted to raise a question.

File: bi_sql_editor/models/bi_sql_view.py
Comment (henrybackman) [inline bi_sql_editor/models/bi_sql_view.py]: Should use `@api.ondelete` instead
- https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-15.0

File: bi_sql_editor/models/bi_sql_view.py
Comment (henrybackman) [inline bi_sql_editor/models/bi_sql_view.py]: Could the `readonly` fields be marked as `readonly` in the view instead?
- https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-17.0

File: bi_sql_editor/views/view_bi_sql_view.xml
Comment (henrybackman) [inline bi_sql_editor/views/view_bi_sql_view.xml:8]: This and other references to `tree` should be changed to `list`
- https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-18.0

**Review by henrybackman — CHANGES_REQUESTED** [review]
Small change requests but overall looks good :+1:

File: bi_sql_editor/models/bi_sql_view.py
Comment (sebalix) [inline bi_sql_editor/models/bi_sql_view.py]: Better to use `sql.SQL` and `sql.Identifier` from `psycopg2` instead to avoid SQL injection, see https://www.psycopg.org/docs/sql.html

File: bi_sql_editor/models/bi_sql_view.py
Comment (sebalix) [inline bi_sql_editor/models/bi_sql_view.py]: Same remark

File: bi_sql_editor/models/bi_sql_view.py
Comment (sebalix) [inline bi_sql_editor/models/bi_sql_view.py]: Same remark

**Review by sebalix — COMMENTED** [review]
LG, small remarks

**Review by lef-adhoc — APPROVED** [review]

**Review by JrAdhoc — APPROVED** [review]
LGTM

**Review by henrybackman — APPROVED** [review]

**Comment by sebalix** [comment]
Hello @thienvh332 , you can drop your `DON'T MERGE` commit, all dependencies have been merged now :) Thanks!

**Comment by thienvh332** [comment]
>  you can drop your `DON'T MERGE` commit, all dependencies have been merged now :) Thanks!

Hi @sebalix ,
Thanks, I updated it.

**Review by sebalix — APPROVED** [review]

**Review by sebalix — COMMENTED** [review]
Getting back on my review, I couldn't create SQL objects on a demo report (on runbot):
image
Can you check this please?

**Comment by thienvh332** [comment]
Hi @sebalix,
The problem comes from my side, sorry about that.

I fixed it. Thanks

**Review by HviorForgeFlow — APPROVED** [review]

**Pull request merged**
